hooks/after_setup/add_host_storage_domain.yml

```diff
@@ -0,0 +1,51 @@
+- name: Include Host vars
+  include_vars: "{{ file_item }}"
+  with_fileglob: "/usr/share/ansible/gluster_inventory.yml"
```

Comment:
with_fileglob does the job. It first check if file exist.
Answering to your NFS or ISCSI question. Inventory file handles that. It will read only when gluster defined. In case of other storage playbook will simply skip.